faunadb-scala/src/test/scala/faunadb/ClientSpec.scala

```diff
@@ -126,6 +126,20 @@ class ClientSpec
     client.query("qux").futureValue should equal (StringV("qux"))
   }
 
+  it should "echo a query value" in {
+    // Run
+    val result = client.query(
+      QueryV(ObjectV("lambda" -> "_", "expr" -> true))
+    ).futureValue
+
+    // Verify
+    result shouldBe a[QueryV]
+    val query = result.asInstanceOf[QueryV]
+    query.lambda("lambda").to[String].get shouldBe "_"
+    query.lambda("expr").to[Boolean].get shouldBe true
+    query.lambda("api_version").toOpt shouldBe defined
```

Comment:
you can assert `api_version` is equal the current api version in the driver

Reply:
also, please add another test/assertion where you create a `QueryV` value with no `api_version` field, and assert the response contains `api_version = 2.12`

Reply:
Thanks for the comments, Marrony! :)

> you can assert `api_version` is equal the current api version in the driver

Yeah, I didn't want to hardcode the API Version in the assertion since it would require us to update this test manually every time we upgrade to a newer version. Also, I'm not sure if it's worthy to make the API Version accessible outside of the `common.Connection` to achieve that either (I remember we discussed not do so for a similar case in the past).

> also, please add another test/assertion where you create a QueryV value with no api_version field, and assert the response contains api_version = 2.12

We would run to the same issues mentioned above, right?

Reply:
@nathaniel-may, since the boundaries between what should be tested in core vs in the drivers are not yet clear, do you think we should include the test cases Marrony mentioned in the drivers as well, or is it good to go for now? Thanks!

Reply:
> We would run to the same issues mentioned above, right?
no, it will always be 2.12

Reply:
I think I'm bit confused about the expected behaviour now...

So, if I create a `QueryV` with no `api_version`, the returned value would always be `2.12` then, even if the current API version in the database is, for instance, `3.0`?

Reply:
that's right, query values with no api_version means it's a legacy value that should be handled correctly even using new drivers